### DISCUSSION TITLE
Problem with handling responses, errors and etc

### DISCUSSION THREAD

**Discussion by Ohyenno** [fastapi/fastapi#11288] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
Files structure below

#main.py
app = FastAPI()
app.include_router(f.router)

#f.py
router = APIRouter()
router.include_router(example_router.router)

#example_router.py
router = APIRouter()
@router.get("/v1/somepath", auth: OAuth2 = Depends())
items = Items()

#Items.py
class with code, exceptions etc
some times could be dependents

#some_dependents.py
lots of code, exceptions etc

#OAuth2 
lots of methods
```

### Description

Hi. I apologize if I am writing in the wrong place. 

After changing the logic for handling errors and exceptions (0.110.0), there is problems with nested routing via @router. 

The raise handlers does not work when accessing routes through APIRouter(). Sometime we can catch errors by using the return JSONResponse or Response (not everywhere, the reason has not yet been found), but nothing works at all in depends. Unfortunately, we have to rollback to the previous build. 

Routing from the root works great as in all of your examples, but it is not convenient way with larger projects. With nesting files and methods error handling is especially difficult to implement (even before). I think it'll great to catch any exceptions through middleware level or smth like that. When an error or response is thrown, we need to stop the request anywhere and return an immediate response. Now we need to send return back to the parent, then do it again and repeat again. In some routing methods we have up to four levels of nesting. it is especially inconvenient when you need to make a http redirect (not RedirectResponse).

Additionally, it would be beneficial if we could throw exceptions in a single class (with HTTPExceptions, Response etc), because any control of errors and responses is one layer. So with this idea any loggers could be attached to them.

And more thing. Along with an exception being thrown, it would also be convenient to have immediate access to the method name, class name or even filename that caused the exceptions.

Thanks for you product!

### Operating System

macOS

### FastAPI Version

0.110.0

### Pydantic Version

2.4.2

### Python Version

Python 3.10.2

**Comment by YuriiMotov** [comment]
Is it still relevant?

Let's close this for now to keep the discussions clean and organized.
If you still need help, feel free to let us know.
